habitat-lab/habitat/datasets/rearrange/navmesh_utils.py

```diff
@@ -10,6 +10,137 @@
 from habitat_sim.physics import VelocityControl
 
 
+def snap_point_is_occluded(
+    target: mn.Vector3,
+    snap_point: mn.Vector3,
+    height: float,
+    sim: habitat_sim.Simulator,
+    granularity: float = 0.2,
+    target_object_id: Optional[int] = None,
+) -> bool:
+    """
+    Uses raycasting to check whether a target is occluded given a navmesh snap point.
+
+    :property target: The 3D position whic should be unoccluded from the snap point.
```

Comment:
typo: which

Reply:
And I imagine this will be gripper's EE location?

Reply:
target is the location you want visibility to (e.g. the target object's location).